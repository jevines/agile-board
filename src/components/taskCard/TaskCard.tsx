import React from "react";
import styled from "styled-components";

export interface Task {
  id: number;
  title: string;
  description: string;
  comments: string[];
}

const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  background: white;
  border-radius: 0.5rem;
  margin: 2rem;
  padding-left: 1rem;
`;
const TaskTitle = styled.h2`
  font-size: 20;
  color: black;
  text-align: left;
`;
const TaskDescription = styled.p`
  font-size: 16;
  color: black;
  text-align: left;
`;
const TaskComment = styled.p`
  font-size: 14;
  color: black;
  text-align: left;
`;

interface TaskFileProps {
  task: Task;
}

const TaskCard = ({ task }: TaskFileProps) => {
  return (
    <Wrapper>
      <TaskTitle>{task.title}</TaskTitle>
      <TaskDescription>{task.description}</TaskDescription>
      {task.comments.map((comment, index) => {
        return <TaskComment key={index}>{comment}</TaskComment>;
      })}
    </Wrapper>
  );
};

export default TaskCard;
