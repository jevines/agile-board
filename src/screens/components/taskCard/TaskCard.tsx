import React from "react";
import styled from "styled-components";

export interface Task {
  title: string;
  description: string;
  comments: string[];
}

const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  background: white;
  border-radius: 0.5rem;
  margin: 0 2rem 0 2rem;
`;
const TaskTitle = styled.h2`
  font-size: 20;
  color: black;
  text-align: left;
`;
const TaskDescription = styled.p`
  font-size: 16;
  color: black;
  text-align: left;
`;
const TaskComment = styled.p`
  font-size: 14;
  color: black;
  text-align: left;
`;

interface TaskFileProps {
  task: Task;
}

const TaskCard = ({ task }: TaskFileProps) => {
  return (
    <Wrapper>
      <TaskTitle>{task.title}</TaskTitle>
      <TaskDescription>{task.description}</TaskDescription>
      {task.comments.map((comment, index) => {
        return <TaskComment key={index}>{comment}</TaskComment>;
      })}
    </Wrapper>
  );
};

export default TaskCard;
