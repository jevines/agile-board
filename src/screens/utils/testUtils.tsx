import React from "react";
import { render } from "@testing-library/react";
import { DragDropContext } from "react-beautiful-dnd";

export const renderWithDragContext = (
  Component: React.ReactChild,
  onDragEndMock = jest.fn()
) => {
  return render(
    <DragDropContext onDragEnd={onDragEndMock}>{Component}</DragDropContext>
  );
};
