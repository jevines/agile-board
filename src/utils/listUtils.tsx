import { Task } from "../components/taskCard/TaskCard";

export const reorderTaskList = (
  list: Task[],
  startIndex: number,
  destinationIndex: number
) => {
  const reorderedList = Array.from(list);
  const [removed] = reorderedList.splice(startIndex, 1);
  reorderedList.splice(destinationIndex, 0, removed);

  return reorderedList;
};

export const moveTask = (
  source: Task[],
  destination: Task[],
  sourceIndex: number,
  destinationIndex: number
) => {
  const sourceList = Array.from(source);
  const destinationList = Array.from(destination);
  const [removed] = sourceList.splice(sourceIndex, 1);

  destinationList.splice(destinationIndex, 0, removed);

  return { sourceList, destinationList };
};
