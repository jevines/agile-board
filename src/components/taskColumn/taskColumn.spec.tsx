import React from "react";
import TaskColumn from "./taskColumn";
import { mockTask } from "../taskCard/TaskCard.spec";
import { renderWithDragContext } from "../../screens/utils/testUtils";

describe("The TaskColumn Component", () => {
  it("should match the snapshot", () => {
    const { container } = renderTaskColumn();

    expect(container.firstChild).toMatchSnapshot();
  });
});

const renderTaskColumn = () => {
  return renderWithDragContext(
    <TaskColumn title="test title" tasks={[mockTask]} droppableId="testId" />
  );
};
