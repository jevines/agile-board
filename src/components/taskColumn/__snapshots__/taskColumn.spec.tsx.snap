// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TaskCard Component should match the snapshot 1`] = `
<section
  class="sc-bdfBwQ fFBvFt"
>
  <h2
    class="sc-gsTCUz gzfVJk"
  >
    mock task title
  </h2>
  <p
    class="sc-dlfnbm bglaea"
  >
    A mock description
  </p>
  <p
    class="sc-hKgILt fLjEpC"
  >
    comment 1
  </p>
  <p
    class="sc-hKgILt fLjEpC"
  >
    comment 2
  </p>
</section>
`;

exports[`The TaskColumn Component should match the snapshot 1`] = `
<section
  class="sc-eCssSg jmQFqn"
>
  <h1
    class="sc-jSgupP dhcVWK"
  >
    test title
  </h1>
  <section
    class="sc-bdfBwQ fFBvFt"
  >
    <h2
      class="sc-gsTCUz gzfVJk"
    >
      mock task title
    </h2>
    <p
      class="sc-dlfnbm bglaea"
    >
      A mock description
    </p>
    <p
      class="sc-hKgILt fLjEpC"
    >
      comment 1
    </p>
    <p
      class="sc-hKgILt fLjEpC"
    >
      comment 2
    </p>
  </section>
</section>
`;
